Automattic/wordpress-activitypub
# Add like handler

Adds basic support for incoming likes (see #235).

Probably needs to go through a couple more iterations before it could be merged! For instance, the ActivityPub plugin should maybe also get a `pre_get_comments` filter, similar to the Webmention plugin, so that these "likes" don't show up among "regular comments." Should be "rather easy" to implement.

But it's a "first draft."

I originally wanted to also add "repost" support but noticed an announce handler already exists. I'm going to have to check that out later. Maybe in a different PR.

Edit: This won't actually _display_, e.g., a facepile. Though it currently is compatible with IndieBlocks' Facepile block. I can imagine you'd want to eventually add a block of your own ...

## Proposed changes:
* Adds a incoming like handler, and extends the existing undo handler to handle undoing of previous likes.

Note: I kept some of the actions from other reaction handlers and also set some comment meta to be able to later on display the like in a "facepile" kind of way. I also set some "dummy" content; feel free to change it in order to have it behave the same as, e.g., the Webmention plugin.

### Other information:


## Testing instructions:






* Upload patched version
* Search for a local URL from, e.g., your Mastodon instance
* Like the corresponding post
* Watch the "like" appear as a new comment with comment type "like"
* Undo like
* Watch the "like" comment move to trash
